server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/MySQLAffectedRowsExceedLimit.java

```diff
@@ -120,14 +122,33 @@ public List<DialectType> getSupportsDialectTypes() {
         return Arrays.asList(DialectType.MYSQL, DialectType.OB_MYSQL);
     }
 
+    /**
+     * MySQL checks the count of value list. Process mode: case1: For INSERT INTO ... VALUES(...)
+     * statements, ODC checks the count of value list. case2: For INSERT INTO ... SELECT ... statements,
+     * ODC runs EXPLAIN statements to get affected rows.
+     *
+     * @param explainSql target sql
+     * @return affected rows
+     */
+    private long getMySqlAffectedRowsByCount(String explainSql, JdbcOperations jdbc) {
+
+        explainSql = explainSql.toLowerCase();
+
+        if (explainSql.contains("select")) {
+            return getMySqlAffectedRowsByExplain(explainSql, jdbc);
+        } else {
+            return explainSql.split("\\),").length;
```

Comment:
The method is not robust.
Check `insert_stmt` in `OBParser.g4`, it contains a lot of syntaxes including '),'

Reply:
Ok, I'll achieve it in a more accurate way.